File: typescript/src/domain/Sku.ts
// ABOUTME: Stock keeping unit identifier with validation
// ABOUTME: Type-safe wrapper for SKU strings

export class Sku {
  private readonly _value: string;

  constructor(value: string) {
    if (!value || value.trim().length === 0) {
      throw new Error('Sku cannot be empty');
    }
    this._value = value.trim();
  }

  get value(): string {
    return this._value;
  }

  equals(other: Sku): boolean {
    return this._value === other._value;
  }

  toString(): string {
    return this._value;
  }
}

File: java/src/main/java/com/promoengine/domain/Sku.java
// ABOUTME: Type-safe wrapper for stock keeping unit identifiers.
// ABOUTME: Ensures SKUs are non-empty strings and provides hashable identity.
package com.promoengine.domain;

public record Sku(String value) {

    public Sku {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("SKU cannot be null or empty");
        }
    }

    @Override
    public String toString() {
        return value;
    }
}
